projects/cm_mcu/clocksynth.c

```diff
@@ -74,10 +74,10 @@ void getClockProgram(int device, char progname_clkdesgid[CLOCK_PROGNAME_REG_NAME
     return;
 
   // grab the semaphore to ensure unique access to I2C controller
```

Comment:
why is this commented out?

Reply:
there is already semaphore take in a larger scope where `getClockProgram` is called. See https://github.com/apollo-lhc/cm_mcu/blob/78ae126eeb029e2c4a4f7a6b1d28e2ab8be7bde0/projects/cm_mcu/InitTask.c#L57

Reply:
Ok, then please remove it and add a comment at the top of the function that you _must_ grab the semaphore before calling this function

Reply:
Done.